Libplanet.Net/Transports/CommunicationFailException.cs

```diff
@@ -36,9 +40,7 @@ public class CommunicationFailException : Exception
         public CommunicationFailException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
-            Peer = info.GetValue(nameof(Peer), typeof(BoundPeer)) is BoundPeer peer
-                ? peer
-                : throw new SerializationException($"{nameof(Peer)} is of an invalid type.");
+            Peer = new BoundPeer(_codec.Decode(info.GetValue<byte[]>(nameof(Peer))));
```

Comment:
Would it be possible that `info.GetValue<byte[]>()` returns null or non-Dictionary bencodex value?

Reply:
It is. I guess its pretty sloppy. Then again, it could be argued whether `SerializationException` is all that necessary in this context (seems like a defensive practice for the sake of the practice itself). 🙄

I'll take another look.

Reply:
Seems like this problem runs a bit deeper than it looks. I think for proper `null` handling, changes to `SerializationInfoExtensions` might be the way to go (although, I'm more of the opinion we need to get rid of `SerializationInfoExtensions`).